scripts/setup_mysql/mysql_docker.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+const exec = require('child_process').exec;
+const path = require('path');
+const os = require('os');
+
+const command = process.argv[2];
+
+const dbAdminPassword = 'super_STRONG_p4ssword';
+const dbUserName = 'admin';
+const dbUserPassword = 'admin';
+const dbName = 'atlasengine';
+const dbPort = 3306;
+const maxRetries = 20;
+
+const dbDockerImageName = 'atlas_engine_mysql';
+const dbContainerName = 'atlas_engine_mysql_container';
+const dbVolumeName = 'atlas_engine_mysql_volume';
+
+let logPath = '/dev/null';
+if (os.platform() === 'win32') {
+  logPath = 'NUL';
+}
+async function sleep(milliseconds) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, milliseconds);
+  });
+}
+
+function runCommand(commandToRun) {
+  return new Promise((resolve, reject) => {
+    exec(commandToRun, {maxBuffer: 2097152}, (error, stdout, stderr) => {
+      if (error !== null) {
+        return reject(error);
+      }
+
+      return resolve(stdout);
+    });
+  });
+}
+
+async function createDbContainer() {
+  const containerId = await runCommand(`\
+    docker run \
+      --detach \
+      --env MYSQL_ROOT_PASSWORD=${dbAdminPassword} \
+      --env MYSQL_USER=${dbUserName} \
+      --env MYSQL_PASSWORD=${dbUserPassword} \
+      --publish ${dbPort}:3306 \
+      --name ${dbContainerName} \
+      --mount source=${dbVolumeName},target=/var/lib/mysql \
+      mysql:latest > ${logPath} \
+    `);
+
+  console.log('Waiting until the Container becomes ready...');
+  process.stdout.write('1');
+  await sleep(1000);
+
+  for (let tryNumber = 2; tryNumber <= maxRetries + 1; tryNumber += 1) {
```

Comment:
Warum fängt die Schleife bei 2 an, statt 1?

```suggestion
  for (let tryNumber = 1; tryNumber <= maxRetries; tryNumber ++) {
```

Reply:
Die Iterationsvariable gillt hier einfach als Anzahl der Sekunden, die bereits auf den Container gewartet wurde.

Da in Zeile 57 schon eine Sekunde gewartet wurde, befindet man sich zum Zeitpunkt der schleife schon in der zweiten Sekunde. Ich kann es aber natürlich auch eben ummodeln.

Reply:
Ja, bitte. 
Es ist nie eine gute Idee, den produktiven Code an die Logausgaben anzupassen. 

Viel mehr sollten sich die Logausgaben am Produktivcode orientieren und geloggte Werte - sofern nötig - in ein Lesbares Format bringen.